Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -768,7 +768,9 @@ public final class PackageBuilder {
                 providers: manifestTarget.providers
             )
         } else if potentialModule.type == .binary {
-            guard let artifact = self.binaryArtifacts.first(where: { $0.path == potentialModule.path }) else {
+            guard let artifact = self.binaryArtifacts.first(where: {
+                $0.path == potentialModule.path || $0.path.basenameWithoutExt == potentialModule.path.basenameWithoutExt
```

Comment:
With the current approach, the base name is the only thing that we can compare to match the artifacts (and that is the price for keeping the public API intact), and that means that the local archive base name must always match the target name. I'm wondering if that is a reasonable restriction (which we would have to reflect in the documentation), or should we try to do something about that?

Reply:
would the whole path be the same, other than the extension?

Reply:
That's a great question, it turned out that the paths were different because the potential module path was determined incorrectly at the earlier step. It is fixed now by handling local archives as a special case of manifest targets with custom paths: https://github.com/apple/swift-package-manager/pull/3746/files#diff-68f661bfe880d4bf7059718964ee104add3839e7e2f5bc8bfd644ab706ff7149R551-R556